prow/github/hmac.go

```diff
@@ -46,3 +79,40 @@ func PayloadSignature(payload []byte, key []byte) string {
 	sum := mac.Sum(nil)
 	return "sha1=" + hex.EncodeToString(sum)
 }
+
+// extractHmacs returns all *valid* HMAC tokens for given repository/organization. It iterates over full tree as necessary (repo->org->global)
+func extractHmacs(repo string, tokenGenerator func() []byte) ([][]byte, error) {
+	t := tokenGenerator()
+	repoToTokenMap := map[string]hmacsForRepo{}
+
+	if err := yaml.Unmarshal(t, &repoToTokenMap); err != nil {
+		//NOTE : To keep backward compatibility, we are going to assume that in case of error, whole while is a single line hmac token.
```

Comment:
Nit: I don't think "NOTE" is useful here. Also, s/while/file

```suggestion
		// To keep backward compatibility, we are going to assume that in case of error
		// the whole file is a single line hmac token.
```